src/components/chips/js/chipsDirective.js

```diff
@@ -67,11 +67,16 @@
    *    displayed when there is at least on item in the list
    * @param {boolean=} readonly Disables list manipulation (deleting or adding list items), hiding
    *    the input and delete buttons
-   * @param {expression} md-on-append An expression that when called expects you to return an object
-   *    representation of the chip input string.
+   * @param {expression} md-on-append An expression of form `myFunction($chip)` that when called
```

Comment:
I think this is not a good API... Partly my fault for the bad naming.
The original intent of md-on-append was to provide a means for the Chips user to build a new object from the input string. It should really have been named something like "chip-builder".
Using the md-on-append as a notification mechanism is troublesome because the String published is often not the final value to be added (esp. since it's expected that on-append will change it), and would be further broken if any other processing is added.

Is md-on-select not sufficient for notifying of new chips?